src/journal-remote/journal-gatewayd.c

```diff
@@ -307,42 +345,99 @@ static int request_parse_range(
         if (!colon)
                 m->cursor = strdup(range);
```

Comment:
If `request_parse_range_entries()` is called twice, `m->cursor` is leaked. Maybe it is better to add `m->cursor = mfree(m->cursor);`

Reply:
Ok ! done.